### PR TITLE
Aggregate server history dropdown

### LINKED ISSUE #255: ODK Aggregate URL History
**Problem description**
Once a change is made on the ODK aggregate URL, I can't get back to the original URL. 
When using multiple servers on one device, I need to remember all the servers I use.

**Proposed Solution**
Could we have a drop down on the Server URL that stores the last 5 URL's used.

### PR DESCRIPTION
**Pull request by grzesiek2010**
This PR is in reference to resolve #255

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java [hunk 1]
```diff
@@ -55,6 +76,7 @@ protected void onPause() {
         }
     }
 
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java]: This is just a way to block Lint so I believe that the new code which uses features of an API level > 9 will result in a hard crash if the preferences screen is opened on devices that run Android versions below Honeycomb. Is that right?

**Review by lognaturel — CHANGES_REQUESTED** [review]
In terms of behavior, this is perfect! The only thing I'd like to explore is whether it's possible to have a slightly bigger down arrow icon. I see you used a material design icon which is great. Is its size just determined by its container and the margin on the image itself?

Can you say a little bit about why you decided to use json for persisting the list rather than, for example, a set (which can be saved directly) or serializing an arraylist? I just want to make sure we've considered the tradeoffs.

File: collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java (hunk 1 above)
Comment (lognaturel) [reply collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java]: This is not necessarily a problem but I do want to understand it. I'm hoping we can address #293 before the next release anyway.

**Comment by grzesiek2010** [comment]
I added bigger icons. 
I decided to use json because the Set interface does not provide any ordering guarantees. I saw this solution too: http://stackoverflow.com/a/7057858/5479029 but it reqires ObjectSerializer class so I think my solution is ok.

File: collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java]: Yes we should wait for #293 an then remove above targetapi.

File: collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java [hunk 2]
```diff
@@ -25,26 +28,44 @@
...
 
+    private static final String KNOWN_URL_LIST = "knownUrlList";
+
     protected EditTextPreference mServerUrlPreference;
     protected EditTextPreference mUsernamePreference;
     protected EditTextPreference mPasswordPreference;
     protected boolean mCredentialsHaveChanged = false;
 
+    private ListPopupWindow mListPopupWindow;
+    private List<String> mUrlList;
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java:66]: Does this need to be a field or can it be a local variable?

Comment (lognaturel) [reply collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java:66]: And actually, same question for mAdapter. Would be better to have local variables wherever possible.

**Comment by lognaturel** [comment]
Icon looks great. I didn't notice this before but on lines 166 and 168 of AggregatePreferencesActivity.java, the latest server URL goes to position 1 rather than 0 which "protects" the default https://opendatakit.appspot.com server. It should not have any special status, though and new URLs should bump it like any other.

I made a note of this inline but also verified it, `mAdapter` and `mUrlList` can both be local variables (without the `m`). Just make sure `mUrlList` is declared final in `onCreate` so you can use it in `onItemClick`.

I know the existing code is not very well organized but when adding new features it would be great to start breaking it up into methods. I'm seeing a good `addUrlToPreferencesList` method and something like `urlDropdownSetup` (might be better to break that one even further).

**Comment by grzesiek2010** [comment]
@lognaturel I made some changes please review again.

**Comment by lognaturel** [comment]
@grzesiek2010 I'm not really sure what happened on CircleCI but I can't rerun the build. Can you please get that passing and then I think this is ready to merge.

**Comment by grzesiek2010** [comment]
@lognaturel it's ready

**Pull request merged**